libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.cpp

```diff
@@ -569,12 +580,9 @@ int ESP8266WiFiGenericClass::hostByName(const char* aHostname, IPAddress& aResul
  */
 void wifi_dns_found_callback(const char *name, CONST ip_addr_t *ipaddr, void *callback_arg)
 {
-    (void) name;
-    if (!_dns_lookup_pending) {
-        return;
-    }
+    (reinterpret_cast<struct arg_callback*>(callback_arg))->called = true;
     if(ipaddr) {
-        (*reinterpret_cast<IPAddress*>(callback_arg)) = IPAddress(ipaddr);
+        (reinterpret_cast<struct arg_callback*>(callback_arg))->addr = ipaddr->addr;

```

Comment:
`(reinterpret_cast<struct arg_callback*>(callback_arg))->addr = addr;`
should work. `IPAddress` can take a `ip_addr_t*` and it is also IPv6 compatible.

Reply:
In my opinion there's something wrong with your reply. In this function "callback_arg" is a pointer to a struct and you're trying to save a IpAddress into a struct pointer, instead of saving it into the "addr" member of the struct.

Reply:
see https://github.com/esp8266/Arduino/blob/7c2e071351fa5db136b685dcc57fd30e12c451c6/cores/esp8266/IPAddress.h#L153 and https://github.com/esp8266/Arduino/blob/7c2e071351fa5db136b685dcc57fd30e12c451c6/cores/esp8266/IPAddress.h#L167 when IPv6 is enabled.

If it does not work, then we can also overload the `operator =`.

Reply:
I modified the line. Is it OK? Or have I misunderstood your message?

Reply:
It is OK.
More generally, you must update all of your `.addr` to use `IPAddress` instead otherwise It will not compile with IPv6.
IPAddress has lots of constructors, but maybe lacks of some overloaded `operator=`.
I'll eventually add them if needed.
Or you can, that would be adding in `IPAddress.h` L152:
```
IPAddress& operator= (const ipv4_addr& fw_addr) { setV4(); v4() = fw_addr.addr; return *this; }
IPAddress& operator= (const ipv4_addr* fw_addr) { setV4(); v4() = fw_addr->addr; return *this; }
...
#if LWIP_IPV6
...
IPAddress& operator= (const ip_addr_t& lwip_addr) { ip_addr_copy(_ip, lwip_addr); return *this; }
IPAddress& operator= (const ip_addr_t* lwip_addr) { ip_addr_copy(_ip, *lwip_addr); return *this; }
```

With them you can 
```
 if(ipaddr) {
        //(*reinterpret_cast<IPAddress*>(callback_arg)) = IPAddress(ipaddr);
        *(*reinterpret_cast<IPAddress*>(callback_arg)) = ipaddr;
  }
```

Reply:
Ok, thank you so much